diff crashes when a rule has no actions

To reproduce:
1. In gmail, create a label "deleteme"
2. In gmail, create a rule that applies label "deleteme" and performs no other actions
3. Delete label "deleteme"
4. Run `gmailctl diff`

This causes the following stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x8e22a8]

goroutine 1 [running]:
github.com/mbrt/gmailctl/pkg/export/api.defaultImporter.importAction(0x0, 0xb82260, 0xc0005ac450, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/pkg/export/api/api_import.go:65 +0x58
github.com/mbrt/gmailctl/pkg/export/api.defaultImporter.importFilter(0xc0007aea20, 0xb82260, 0xc0005ac450, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/pkg/export/api/api_import.go:48 +0x8e
github.com/mbrt/gmailctl/pkg/export/api.defaultImporter.Import(0xc00058a200, 0x3e, 0x3f, 0xb82260, 0xc0005ac450, 0x0, 0x7fd80111a000, 0xc0001da5a0, 0xc0001f6700, 0xc0001ec0d8)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/pkg/export/api/api_import.go:34 +0xeb
github.com/mbrt/gmailctl/pkg/api.(*gmailAPI).ListFilters(0xc000220760, 0xc000220760, 0x0, 0x0, 0xc00065cec8, 0x8)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/pkg/api/api.go:42 +0x1bc
github.com/mbrt/gmailctl/cmd/gmailctl/cmd.diff(0xc000024440, 0x36, 0xc000024440, 0x36)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/cmd/gmailctl/cmd/diff_cmd.go:54 +0x18e
github.com/mbrt/gmailctl/cmd/gmailctl/cmd.glob..func3(0x11ee1a0, 0x1219318, 0x0, 0x0)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/cmd/gmailctl/cmd/diff_cmd.go:30 +0x42
github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra.(*Command).execute(0x11ee1a0, 0x1219318, 0x0, 0x0, 0x11ee1a0, 0x1219318)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x11eeb20, 0x0, 0xac397b, 0x2d)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra/command.go:852 +0x2c0
github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/vendor/github.com/spf13/cobra/command.go:800
github.com/mbrt/gmailctl/cmd/gmailctl/cmd.Execute()
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/cmd/gmailctl/cmd/root_cmd.go:30 +0x32
main.main()
        /usr/local/google/home/lutzky/go/src/github.com/mbrt/gmailctl/cmd/gmailctl/main.go:6 +0x20
```

Instead, gmailctl should probably just ignore that rule in the importer. The gmail UI attempts to block you from creating rules that have no actions.

--- a/cmd/gmailctl/cmd/apply_cmd.go
+++ b/cmd/gmailctl/cmd/apply_cmd.go
@@ -54,12 +54,12 @@ func apply(path string, interactive bool) error {
 		return configurationError(errors.Wrap(err, "cannot connect to Gmail"))
 	}
 
-	upstreamFilters, err := gmailapi.ListFilters()
+	upstream, err := upstreamFilters(gmailapi)
 	if err != nil {
-		return errors.Wrap(err, "cannot get filters from Gmail")
+		return err
 	}
 
-	diff, err := filter.Diff(upstreamFilters, parseRes.filters)
+	diff, err := filter.Diff(upstream, parseRes.filters)
 	if err != nil {
 		return errors.New("cannot compare upstream with local filters")
 	}

--- a/cmd/gmailctl/cmd/diff_cmd.go
+++ b/cmd/gmailctl/cmd/diff_cmd.go
@@ -51,12 +51,12 @@ func diff(path string) error {
 		return configurationError(errors.Wrap(err, "cannot connect to Gmail"))
 	}
 
-	upstreamFilters, err := gmailapi.ListFilters()
+	upstream, err := upstreamFilters(gmailapi)
 	if err != nil {
-		return errors.Wrap(err, "cannot get filters from Gmail")
+		return err
 	}
 
-	diff, err := filter.Diff(upstreamFilters, parseRes.filters)
+	diff, err := filter.Diff(upstream, parseRes.filters)
 	if err != nil {
 		return errors.New("cannot compare upstream with local filters")
 	}

--- a/cmd/gmailctl/cmd/edit_cmd.go
+++ b/cmd/gmailctl/cmd/edit_cmd.go
@@ -181,12 +181,12 @@ func applyEdited(path, originalPath string, gmailapi api.GmailAPI) error {
 	// This forces labels to be re-fetched, in case something has changed
 	// between edit retries.
 	_, _ = gmailapi.LabelMap()
-	upstreamFilters, err := gmailapi.ListFilters()
+	upstream, err := upstreamFilters(gmailapi)
 	if err != nil {
-		return errors.Wrap(err, "cannot get filters from Gmail")
+		return err
 	}
 
-	diff, err := filter.Diff(upstreamFilters, parseRes.filters)
+	diff, err := filter.Diff(upstream, parseRes.filters)
 	if err != nil {
 		return errors.New("cannot compare upstream with local filters")
 	}

--- a/cmd/gmailctl/cmd/upstream.go
+++ b/cmd/gmailctl/cmd/upstream.go
@@ -1,0 +1,21 @@
+package cmd
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/mbrt/gmailctl/pkg/api"
+	"github.com/mbrt/gmailctl/pkg/filter"
+)
+
+func upstreamFilters(gmailapi api.GmailAPI) (filter.Filters, error) {
+	f, err := gmailapi.ListFilters()
+	if err != nil {
+		if len(f) == 0 {
+			return f, errors.Wrap(err, "cannot get filters from Gmail")
+		}
+		// We have some filters, let's work with what we have and issue a warning.
+		stderrPrintf("Warning: Error getting one or more filters from Gmail: %sThey will be ignored in the diff.\n", err)
+		return f, nil
+	}
+	return f, nil
+}

--- a/pkg/export/api/api_import.go
+++ b/pkg/export/api/api_import.go
@@ -62,6 +62,9 @@ func (di defaultImporter) importFilter(gf *gmailv1.Filter, lmap LabelMap) (filte
 
 func (di defaultImporter) importAction(action *gmailv1.FilterAction, lmap LabelMap) (filter.Actions, error) {
 	res := filter.Actions{}
+	if action == nil {
+		return res, errors.New("empty action")
+	}
 	if err := di.importAddLabels(&res, action.AddLabelIds, lmap); err != nil {
 		return res, err
 	}
@@ -135,11 +138,14 @@ func (di defaultImporter) importCategory(labelID string) gmail.Category {
 	}
 }
 
-func (di defaultImporter) importCriteria(action *gmailv1.FilterCriteria) (filter.Criteria, error) {
+func (di defaultImporter) importCriteria(criteria *gmailv1.FilterCriteria) (filter.Criteria, error) {
+	if criteria == nil {
+		return filter.Criteria{}, errors.New("empty criteria")
+	}
 	return filter.Criteria{
-		From:    action.From,
-		To:      action.To,
-		Subject: action.Subject,
-		Query:   action.Query,
+		From:    criteria.From,
+		To:      criteria.To,
+		Subject: criteria.Subject,
+		Query:   criteria.Query,
 	}, nil
 }

--- a/pkg/export/api/api_import_test.go
+++ b/pkg/export/api/api_import_test.go
@@ -141,3 +141,46 @@ func TestImportLabels(t *testing.T) {
 	_, err = DefaulImporter().Import(filters, lmap)
 	assert.NotNil(t, err)
 }
+
+func TestImportBad(t *testing.T) {
+	// Importing filters with missing pieces doesn't cause crashes.
+	filters := []*gmailv1.Filter{
+		{
+			Action: &gmailv1.FilterAction{
+				AddLabelIds: []string{labelIDTrash},
+			},
+			Criteria: nil,
+		},
+		{
+			Action: &gmailv1.FilterAction{
+				AddLabelIds: []string{labelIDTrash},
+			},
+			Criteria: &gmailv1.FilterCriteria{
+				From: "foo@bar.com",
+			},
+		},
+	}
+	imported, err := DefaulImporter().Import(filters, emptyLabelMap())
+	assert.NotNil(t, err)
+	assert.Len(t, imported, 1)
+
+	filters = []*gmailv1.Filter{
+		{
+			Action: nil,
+			Criteria: &gmailv1.FilterCriteria{
+				From: "foo@bar.com",
+			},
+		},
+		{
+			Action: &gmailv1.FilterAction{
+				AddLabelIds: []string{labelIDTrash},
+			},
+			Criteria: &gmailv1.FilterCriteria{
+				From: "foo@bar.com",
+			},
+		},
+	}
+	imported, err = DefaulImporter().Import(filters, emptyLabelMap())
+	assert.NotNil(t, err)
+	assert.Len(t, imported, 1)
+}

